sound/soc/sof/loader.c

```diff
@@ -129,6 +130,82 @@ int snd_sof_fw_parse_ext_data(struct snd_sof_dev *sdev, u32 bar, u32 offset)
 }
 EXPORT_SYMBOL(snd_sof_fw_parse_ext_data);
 
+static size_t snd_sof_ext_man_size(const struct firmware *fw)
+{
+	const struct ext_man_header *head = (void *)fw->data;
+
+	if (fw->size < sizeof(*head))
+		return 0;
+
+	/*
+	 * When fw points to extended manifest,
+	 * then first u32 must be equal EXT_MAN_MAGIC_NUMBER.
+	 */
+	if (head->magic == EXT_MAN_MAGIC_NUMBER)
+		return head->full_size;
+
+	return 0;
```

Comment:
can this happen or is this an error case?

Reply:
It's a case for FW without ext_man